solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQuery.java

```diff
@@ -127,35 +128,33 @@ public Weight createWeight(IndexSearcher searcher, boolean needsScores, float bo
    * This is the weight for the main solr query in the LTRQuery. The only thing
    * this really does is have an explain using the reRankQuery.
    */
-  public class LTRWeight extends Weight {
+  public class LTRWeight extends FilterWeight {
     private final Rescorer reRankRescorer;
-    private final Weight mainWeight;
     private final IndexSearcher searcher;
 
     public LTRWeight(IndexSearcher searcher, Weight mainWeight,
         Rescorer reRankRescorer) throws IOException {
-      super(LTRQuery.this);
+      super(LTRQuery.this,mainWeight);
       this.reRankRescorer = reRankRescorer;
-      this.mainWeight = mainWeight;
       this.searcher = searcher;
     }
 
     @Override
     public Explanation explain(LeafReaderContext context, int doc)
         throws IOException {
-      final Explanation mainExplain = mainWeight.explain(context, doc);
+      final Explanation mainExplain = super.explain(context, doc);
```

Comment:
minor/style: `in.explain` instead of `super.explain` would be the same behaviour (I think) but avoid one method call.